src/int128_impl.h

```diff
@@ -0,0 +1,18 @@
+#ifndef SECP256K1_INT128_IMPL_H
+#define SECP256K1_INT128_IMPL_H
+
+#include "util.h"
+
+#include "int128.h"
+
+#if defined(SECP256K1_WIDEMUL_INT128)
+#  if defined(SECP256K1_INT128_NATIVE)
+#    include "int128_native_impl.h"
+#  elif defined(SECP256K1_INT128_STRUCT)
+#    include "int128_struct_impl.h"
+#  else
+#    error "Please select int128 implementation"
+#  endif
+#endif
```

Comment:
Although this file should never be included when `SECP256K1_WIDEMUL_INT128` is not defined, we should have an error message here just in case that situation arises.  Otherwise confusing error messages happen.

Reply:
This file is actually included unconditionally in `secp256k1.c`. (One could argue that this is a bad idea, but the nice thing is that it keeps the preprocessor 
 logic